learning/tour-of-beam/frontend/integration_test/welcome_page_test.dart

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:get_it/get_it.dart';
+import 'package:integration_test/integration_test.dart';
+import 'package:playground_components_dev/playground_components_dev.dart';
+import 'package:tour_of_beam/cache/content_tree.dart';
+import 'package:tour_of_beam/cache/sdk.dart';
+import 'package:tour_of_beam/main.dart' as app;
+import 'package:tour_of_beam/state.dart';
+
+import 'common/common_finders.dart';
+
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+  testWidgets('mode toggle', (wt) async {
+    await app.main();
+    await wt.pumpAndSettle();
+    await checkToggleBrightnessMode(wt);
+    await _checkSdksLoadedCorrectly(wt);
+    await _checkSwitchingSdkWorksCorrectly(wt);
+  });
+}
+
+Future<void> _checkSdksLoadedCorrectly(WidgetTester wt) async {
+  //The source of truth of loaded sdks
+  final sdkCache = GetIt.instance.get<SdkCache>();
+
+  final sdks = sdkCache.getSdks();
+  for (final sdk in sdks) {
+    expect(
+      find.outlinedButtonWithText(sdk.title),
+      findsOneWidget,
+    );
+
+    expect(
+      find.dropdownMenuItemWithText(sdk.title),
+      findsNothing,
+    );
+  }
+
+  var button = wt.widget<ElevatedButton>(find.startTourButton());
+  expect(button.onPressed, isNull);
+
+  await wt.tapAndSettle(find.outlinedButtonWithText(sdks[0].title));
+
+  button = wt.widget<ElevatedButton>(find.startTourButton());
+  expect(button.onPressed, isNotNull);
+
+  await wt.tapAndSettle(find.sdkDropdwn());
+
+  for (final sdk in sdks) {
+    expect(
+      find.dropdownMenuItemWithText(sdk.title),
+      findsNWidgets(2), //DropdownButton widget peculiarity
+    );
```

Comment:
Is there a way to check that there is only one visible widget with the given text?